### ISSUE TITLE
The environment you requested was unavailable

### ISSUE THREAD

**Issue by rahuldpi** [gemini-testing/gemini#688]
I was trying to run the test using gemini on firefox (v50.0.2), although the firefox browser opened successfully, it didn't open the url in the address bar. Then in the terminal I saw this error message:

> Cannot launch browser firefox: [init({"browserName":"firefox"})] The
> environment you requested was unavailable.

Command:

> sudo gemini update test.js

Code:

    module.exports = {
        rootUrl: 'http://example.com',
        gridUrl: 'http://127.0.0.1:4444/wd/hub',
    
        browsers: {
            chrome: {
               desiredCapabilities: {
                  browserName: 'chrome'
               }
            },
    		'firefox-latest': {
    			desiredCapabilities: {
                    browserName: "firefox"
                }
    		}
        }
    };

The test runs successfully on chrome browser. I'm running MacOSX Sierra.

**Comment by sipayRT** [comment]
hi. there was broken changes with FF driver. And `wd` tool doesn't support it yet. Try to use lower version, pls (for example, 46)

**Comment by sipayRT** [comment]
the same - https://github.com/gemini-testing/gemini/issues/643

**Comment by rahuldpi** [comment]
The same issue also exist on Safari Version 10.0.1.

**Comment by leonsabr** [comment]
@sipayRT Hi guys, are you going to do something about it? Firefox 48+ is no supported by gemini.

**Comment by florinalex24** [comment]
@sipayRT  Hi guys, is there any chance to fix this issue in the near future, as Firefox has reached version 55?

**Comment by sipayRT** [comment]
main problem it's `wd` old version. We are going to drop support node@4.x and update `wd` to the latest version

**Comment by bobanko** [comment]
@sipayRT is there any ticket and ETA for this issue with `wd` version?

**Comment by sipayRT** [comment]
unfortunately firefox still broken in wd - https://github.com/admc/wd/issues/495

**Comment by campside** [comment]
Chrome is also broken?

**Comment by sipayRT** [comment]
no. Chrome should work fine

**Comment by campside** [comment]
`module.exports = {
    rootUrl: 'http://yandex.ru',
    gridUrl: 'http://127.0.0.1:4444/wd/hub',

    browsers: {
        chrome: {
            desiredCapabilities: {
                browserName: 'chrome'
            }
        }
    }
};`

`gemini.suite('yandex-search', (suite) => {
    suite.setUrl('/')
        .setCaptureElements('.home-logo')
        .capture('plain');
});`

`gemini test --reporter flat`

I get http://prntscr.com/hjvefz
and then: http://prntscr.com/hjvern

**Comment by DudaGod** [comment]
@campside hello.
Are you sure that you [run selenium standalone](https://github.com/gemini-testing/gemini#running-tests) before running tests?

**Comment by campside** [comment]
Yes, chrome would not start at all otherwise, http://prntscr.com/hjw0c0

**Comment by sipayRT** [comment]
can you provide chrome and chrome-driver(from selenium-standalone) versions?

**Comment by campside** [comment]
Latest available: 62.0.3202.94 64bit
selenium-standalone, also latest available through install: http://prntscr.com/hk7jvz
MacOs: 10.12.6

**Comment by sipayRT** [comment]
hm...
```
MacOs: 10.12.6
node@8.4.0
gemini@5.0.0
chrome 63.0.3239.84
selenium-standalone@6.12.0 with 2.33-x64-chromedriver
```

works fine for me

**Comment by campside** [comment]
will check everything one more time

**Comment by campside** [comment]
same configuration, reinstalled moduled, same thing: green screen on running chrome then failure.

**Comment by campside** [comment]
We have found the problem, we added: **calibrate: false** in .gemini.js

**Comment by sipayRT** [comment]
but it's wrong solution - gemini should work with calibrate too

**Comment by campside** [comment]
We have not found another way to run it

**Comment by betalb** [comment]
Is it possible in this case to update README and reflect the fact, that Firefox support is limited by older browsers?

**Comment by Kenith** [comment]
I have build a docker image which support firefox 47.0.1, you could have a try here: https://github.com/Kenith/ta-visual-lib, and you could also get the samples.

Hope could be the help

**Comment by Hypnosphi** [comment]
Looks like this wd library is abandoned by maintainers. There is an open PR fixing that with no reaction for half a year
https://github.com/admc/wd/pull/494

How do you feel about considering alternatives (webdriver.io or something else)?

**Comment by generalov** [comment]
@Hypnosphi, the latest commit in `wd` repo was 11 days ago. Firefox example works fine at http://webdriver.io/guide.html.  The issue is out there.

**Comment by Hypnosphi** [comment]
> the latest commit in wd repo was 11 days ago

OK, I'll try to summon the maintainer in that PR

> Firefox example works fine at http://webdriver.io/guide.html
Yes, that's on of the reasons I've suggested webdriver.io as an alternative

**Comment by Hypnosphi** [comment]
Gemini still fails for me with FF 57 and wd 1.10.1:

```
Cannot launch browser firefox:
Could not calibrate. This could be due to calibration page has failed to open properly.
```

**Comment by littlstonee** [comment]
#928

**Comment by littlstonee** [comment]
Please have a look at #928, seems similar.
